core/include/macros/math.h

```diff
@@ -31,7 +31,7 @@ extern "C" {
                               unsigned int: 1,           \
                               unsigned long: 1,          \
                               unsigned long long: 1,     \
-                              default: (long long)(a) < 0 ? -1 : 1)
+                              default: (signed long)(a) < 0 ? -1 : 1)
```

Comment:
shouldn't this be "signed long long" ?

Reply:
I thought so too, but when I would do that I'd still get the error.

Reply:
Could you handle `long long` as special case to avoid loosing precision on the `default`? Otherwise we may end up with false results for 64 bit numbers.

Reply:
```suggestion
                              default: (0LL < ((long long) a)) ? (-1) : (1) )
```

worked for me

i checked and found `long long` to be longer than `unsigned long` (4 vs 8 bytes)

Reply:
hm this will make zero negative signed, but I hope that won't cause any issues…

Reply:
no it is wrong i missed turning the < around

Reply:
Yea I figured. Still `a > 0` will be false and return -1 for `a == 0`.

Reply:
the compiler also fails for uint32_t

Reply:
doesn't the _generics system know about typedefs?